src/Ledger/Conway/Specification/Ratify.lagda

```diff
@@ -243,110 +230,331 @@ designed to do so.
+system these functions calculate whether a governance action is
+ratified by the corresponding body.
+
+\subsubsection{SPO}
+
+\begin{figure*}[!ht]
+\begin{AgdaMultiCode}
 \begin{code}
-actualVotes  : RatifyEnv → PParams → CCData → GovActionType
-             → (GovRole × Credential ⇀ Vote) → (VDeleg ⇀ Vote)
-actualVotes Γ pparams cc gaTy votes
-  =   mapKeys (credVoter CC) actualCCVotes  ∪ˡ actualPDRepVotes gaTy
-  ∪ˡ  actualDRepVotes                       ∪ˡ actualSPOVotes gaTy
+acceptedBySPO
+  : RatifyEnv
+  → EnactState
+  → GovActionState
+  → Type
+acceptedBySPO Γ eSt gaSt = (acceptedStake /₀ totalStake) ≥ t
   where
 \end{code}
 \begin{code}[hide]
-  open RatifyEnv Γ
-  open PParams pparams
+    open EnactState eSt using (cc; pparams)
+    open RatifyEnv Γ
+    open StakeDistrs stakeDistrs
+    open GovActionState gaSt
+    open GovVotes votes using (gvSPO)
 \end{code}
 \begin{code}
-  roleVotes : GovRole → VDeleg ⇀ Vote
-  roleVotes r = mapKeys (uncurry credVoter) (filter (λ (x , _) → r ≡ proj₁ x) votes)
+    castedVotes : KeyHash ⇀ Vote
+    castedVotes = gvSPO
 
-  activeDReps = dom (filter (λ (_ , e) → currentEpoch ≤ e) dreps)
-  spos  = filterˢ IsSPO (dom (stakeDistr stakeDistrs))
+    defaultVote : KeyHash → Vote
+    defaultVote kh = case lookupᵐ? pools kh of
+\end{code}
+\begin{code}[hide]
+      λ where
 \end{code}
 \begin{code}
-
-  getCCHotCred : Credential × Epoch → Maybe Credential
-  getCCHotCred (c , e) = if currentEpoch > e then nothing
-    else case lookupᵐ? ccHotKeys c of
+      nothing   → Vote.no
+      (just  p) → case lookupᵐ? delegatees (StakePoolParams.rewardAccount p) , gaType action of
 \end{code}
 \begin{code}[hide]
-      λ where
+             λ where
 \end{code}
 \begin{code}
-        (just (just c'))  → just c'
-        _                 → nothing -- no hot key or resigned
+             (_                       , TriggerHF)     → Vote.no
+             (just vDelegNoConfidence , NoConfidence)  → Vote.yes
+             (just vDelegAbstain      , _           )  → Vote.abstain
+             _                                         → Vote.no
+
+    actualVotes : KeyHash ⇀ Vote
+    actualVotes = castedVotes ∪ˡ mapFromFun defaultVote (dom stakeDistrPools)
+
+    t : ℚ
+    t = maybe id 0ℚ (threshold (proj₁ pparams) (proj₂ <$> (proj₁ cc)) action SPO)
+
+    acceptedStake totalStake : Coin
+    acceptedStake  = ∑[ x ← stakeDistrPools ∣ actualVotes ⁻¹ Vote.yes                          ] x
+    totalStake     = ∑[ x ← stakeDistrPools ∣ dom (actualVotes ∣^ (❴ Vote.yes ❵ ∪ ❴ Vote.no ❵)) ] x
+\end{code}
+\end{AgdaMultiCode}
+\caption{Vote counting for SPOs}
+\label{fig:defs:ratify-acceptedbyspo}
+\end{figure*}
+
+Cref{fig:defs:ratify-acceptedbyspo}, defines the predicate
+\AgdaFunction{acceptedBySPO}, which uses the following auxiliary
+definitions:
+%
+\begin{itemize}
+  \item \AgdaFunction{castedVotes}: This map contains the votes that
+  have been casted by members of the SPO body and have been collected
```

Comment:
TIL that casted is also correct, I would have thought it had to be cast

Reply:
It seems "cast' is the correct form of the past tense of "cast" and "casted" is not correct.